client/src/app/api/models.ts

```diff
@@ -90,6 +90,10 @@ export interface IdRef {
   id: number;
 }
 
+export interface RefWithRole<RoleType = string> extends Ref {
```

Comment:
Not sure if it's beneficial (long term) to create another type of `Ref`. This requires special handling all over the code base - i.e. in methods like `toRef` which narrow down to pure Ref (in Konveyor almost everything is RefLike). Consider the options:
1. extend  `Ref` - since the prop is optional this should work with the existing code base (but you need to handle all the code that converts the Refs)
2. switch to composition i.e. ` RefWithRole = { ref: Ref, role: string}`. If the type is used only locally the the overhead is minimal and type system will help to avoid bugs.